torch/_inductor/utils.py

```diff
@@ -1665,11 +1667,31 @@ def can_use_tma(*matrices: IRNode, add_guards: bool = False) -> bool:
     def _aligned(expr_bytes: Union[int, sympy.Expr]) -> bool:
         return V.graph.sizevars.statically_known_multiple_of(expr_bytes, TMA_ALIGNMENT)
 
-    def _is_tma_compatible(x: IRNode) -> bool:
-        sizes = x.get_size()
-        strides = x.get_stride()
+    def _is_tma_compatible_layout(layout: Optional[Layout]) -> bool:
+        if layout is None:
+            return True
+        sizes = layout.size
+        strides = layout.stride
+        dtype = layout.dtype
+
+        # TODO: Check Base pointer 16-byte aligned for output?
```

Comment:
Please let me know if you have thoughts on how I can check this. Also happy to do this as a followup.

Reply:
Check if `layout.offset` is statically divisible by 16